### PR TITLE
[SCB-543]optimize registry procedure

### PR DESCRIPTION
**Pull request by mt-monster**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
@liubao68 @laijianbin @wujimin @WillemJiang 
In one case of csp upgrade scene 1: successfully registry microsevice at first , 2: then delete the microservice static information but instance is still registrying , the result is registry instance in hanging on because sc has no microservice static information.so in order to optimize this registry halt , i put registry instance procedure to registry microservice again so let the registry can pass

### CODE REVIEW

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/event/InstanceRegistryFailedEvent.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.apache.servicecomb.serviceregistry.task.event;
+
+/**
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/event/InstanceRegistryFailedEvent.java]: template is not correct?

Comment (mt-monster) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/event/InstanceRegistryFailedEvent.java]: ok.fix it right now

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java [hunk 2]
```diff
@@ -52,6 +53,11 @@ public void onMicroserviceInstanceHeartbeatTask(MicroserviceInstanceHeartbeatTas
     }
   }
 
+  @Subscribe
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java]: it's better to change task controller
when failed to register instance, then change status, next time, task will run from register microservice

Comment (mt-monster) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java]: how to change task controller ?

Comment (liubao68) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java]: Just set 
this.taskStatus = TaskStatus.READY;
this.registered = false;

like onMicroserviceInstanceHeartbeatTask

Comment (mt-monster) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java]: get it

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceInstanceRegisterTask.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
  */
 package org.apache.servicecomb.serviceregistry.task;
 
+import org.apache.servicecomb.foundation.common.event.EventManager;
```
Comment (wujimin) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceInstanceRegisterTask.java]: import bu not used

**Pull request merged**